### PR TITLE
[XamlC] supports enum and consts in x:Static

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

Since 1.3.3, x:Static markups are evaluated at compile time and compiled instead of being resolved by reflection at runtime. Some tests weren't correctly running with XamlC and so some edge cases like `enum` or `const` values weren't correctly handled.

This awesome PR fixes that. And the broken tests too.

### Bugs Fixed ###

- http://forums.xamarin.com/discussion/comment/222060/#Comment_222060

### Behavioral Changes ###

This makes the world a better place.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs [hunk 1]
```diff
@@ -32,24 +32,60 @@ public IEnumerable<Instruction> ProvideValue(IElementNode node, ModuleDefinition
...
+					if (fieldDef.Constant is long)
+						return new [] { Instruction.Create(OpCodes.Ldc_I8, (long)fieldDef.Constant) };
+					if (fieldDef.Constant is ulong)
+						return new [] { Instruction.Create(OpCodes.Ldc_I8, (ulong)fieldDef.Constant) };
+					if (fieldDef.Constant is uint)
+						return new [] { Instruction.Create(OpCodes.Ldc_I4, (uint)fieldDef.Constant) };
+					//everything else will cast just fine to an int
+					return new [] { Instruction.Create(OpCodes.Ldc_I4, (int)fieldDef.Constant) };
+				}
 			}
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs]: is it HasConstant() and is not a predefined type or an enum, I don't know what it could be. So I don't do anything :)

File: Xamarin.Forms.Xaml.UnitTests/StaticExtensionTests.cs [hunk 2]
```diff
@@ -1,72 +0,0 @@
...
-using NUnit.Framework;
-using System.Xml;
-
-using Xamarin.Forms.Core.UnitTests;
-using System.Reflection;
-
-namespace Xamarin.Forms.Xaml.UnitTests
-{
-	[TestFixture]
-	public class StaticExtensionTests : BaseTestFixture
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml.UnitTests/StaticExtensionTests.cs]: All those tests are now ported to be executed with both XamlC on and off

File: Xamarin.Forms.Xaml.UnitTests/XStatic.xaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml.UnitTests/XStatic.xaml]: Can you spot the difference ?

File: Xamarin.Forms.Xaml/MarkupExtensions/StaticExtension.cs [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 namespace Xamarin.Forms.Xaml
 {
-	[ContentProperty("Member")]
+	[ContentProperty(nameof(Member))]
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml/MarkupExtensions/StaticExtension.cs]: that's beautiful

File: Xamarin.Forms.Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs [hunk 5]
```diff
@@ -32,24 +32,60 @@ public IEnumerable<Instruction> ProvideValue(IElementNode node, ModuleDefinition
 			var propertyDef = GetPropertyDefinition(typeRef, membername, module);
 
 			if (fieldRef == null && propertyDef == null)
-				throw new XamlParseException(Format("x:Static: unable to find a public static field or property named {0} in {1}", membername, typename), node as IXmlLineInfo);
+				throw new XamlParseException($"x:Static: unable to find a public static field, static property, const or enum value named {membername} in {typename}", node as IXmlLineInfo);
 
+			var fieldDef = fieldRef?.Resolve();
```
Comment (campersau) [inline Xamarin.Forms.Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs]: Why not put this inside of the `if` below?
